### ISSUE TITLE
Provide a way to configure connection and readTimeout

### ISSUE THREAD

**Issue by aozarov** [GoogleCloudPlatform/gcloud-java#232]
This should be a setting in [`ServiceOptions`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java]).

Once we get it from the user we can set it by  the [`HttpRequestInitializer`](https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/HttpRequestInitializer).

**Comment by aozarov** [comment]
Note that currently the default values are:

connectTimeout - 20 seconds.
readTimeout - 20 seconds.

This was taken from [here](http://grepcode.com/file/repo1.maven.org/maven2/com.google.http-client/google-http-client/1.20.0/com/google/api/client/http/HttpRequest.java#144).

**Comment by mziccard** [comment]
After merging #236, oraclejdk8 build failed with.
```
Tests run: 21, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 55.846 sec <<< FAILURE! - in com.google.gcloud.storage.ITStorageTest
testPostSignedUrl(com.google.gcloud.storage.ITStorageTest)  Time elapsed: 20.011 sec  <<< ERROR!
com.google.gcloud.storage.StorageException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.read(SocketInputStream.java:150)
	at java.net.SocketInputStream.read(SocketInputStream.java:121)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:961)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:918)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:703)
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:647)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1535)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at com.google.api.client.http.javanet.NetHttpResponse.<init>(NetHttpResponse.java:37)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:94)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:972)
	at com.google.api.client.googleapis.media.MediaHttpUploader.executeCurrentRequestWithoutGZip(MediaHttpUploader.java:545)
	at com.google.api.client.googleapis.media.MediaHttpUploader.executeCurrentRequest(MediaHttpUploader.java:562)
	at com.google.api.client.googleapis.media.MediaHttpUploader.directUpload(MediaHttpUploader.java:360)
	at com.google.api.client.googleapis.media.MediaHttpUploader.upload(MediaHttpUploader.java:334)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:427)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at com.google.gcloud.spi.DefaultStorageRpc.create(DefaultStorageRpc.java:137)
	at com.google.gcloud.storage.StorageImpl$3.call(StorageImpl.java:142)
	at com.google.gcloud.storage.StorageImpl$3.call(StorageImpl.java:139)
	at com.google.gcloud.RetryHelper.doRetry(RetryHelper.java:181)
	at com.google.gcloud.RetryHelper.runWithRetries(RetryHelper.java:247)
	at com.google.gcloud.RetryHelper.runWithRetries(RetryHelper.java:237)
	at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:139)
	at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:124)
	at com.google.gcloud.storage.ITStorageTest.testPostSignedUrl(ITStorageTest.java:453)
```

**Comment by aozarov** [comment]
Yes, re-opened #220, hoping #234 would fix it.

**Comment by aozarov** [comment]
Fixed by #234
